### ISSUE TITLE
Latest build fails to install - need to downgrade in order to build the DB first

### ISSUE THREAD

**Issue by bertrandgorge** [bitnami/bitnami-docker-discourse#201]
**Description**

A vanilla install with the latest 2-debian-10 tag failed to install. The script stops during db:migrate, apparently because of the plugins (not an expert of Discourse, just an assumption).

We restarted from scratch, same docker-compose, but this time with the 2.6.2-debian-10-r13 tag. The script completed successfully.

We then upgraded to the latest version again, while keeping the DB, and the setup completed successfully this time.

**Steps to reproduce the issue:**

1. docker-compose up -d with tag 2-debian-10
2. fail

**Describe the results you received:**

The db:migrate step fails with the following error (in the docker logs, the container just stops during the `discourse 17:28:15.55 INFO ==> Populating database` phase) :

```
root@fef60997239b:/opt/bitnami/discourse# bundle exec rake db:migrate
** PLUGIN FAILURE **
You are unable to build Discourse due to this error during plugin
initialization:
uninitialized constant Plugin::Instance::DiscourseDev
/opt/bitnami/discourse/lib/plugin/instance.rb:98:in `block in find_all'
/opt/bitnami/discourse/lib/plugin/instance.rb:86:in `tap'
/opt/bitnami/discourse/lib/plugin/instance.rb:86:in `find_all'
lib/discourse.rb:243:in `activate_plugins!'
/opt/bitnami/discourse/config/application.rb:312:in `block in <class:Application>'
/opt/bitnami/discourse/lib/plugin_initialization_guard.rb:5:in `plugin_initialization_guard'
/opt/bitnami/discourse/config/application.rb:311:in `<class:Application>'
/opt/bitnami/discourse/config/application.rb:73:in `<module:Discourse>'
/opt/bitnami/discourse/config/application.rb:72:in `<top (required)>'
/opt/bitnami/discourse/Rakefile:7:in `require'
/opt/bitnami/discourse/Rakefile:7:in `<top (required)>'
/opt/bitnami/discourse/vendor/bundle/ruby/2.7.0/gems/rake-13.0.3/lib/rake/rake_module.rb:29:in `load'
/opt/bitnami/discourse/vendor/bundle/ruby/2.7.0/gems/rake-13.0.3/lib/rake/rake_module.rb:29:in `load_rakefile'
/opt/bitnami/discourse/vendor/bundle/ruby/2.7.0/gems/rake-13.0.3/lib/rake/application.rb:703:in `raw_load_rakefile'
/opt/bitnami/discourse/vendor/bundle/ruby/2.7.0/gems/rake-13.0.3/lib/rake/application.rb:104:in `block in load_rakefile'
/opt/bitnami/discourse/vendor/bundle/ruby/2.7.0/gems/rake-13.0.3/lib/rake/application.rb:186:in `standard_exception_handling'
/opt/bitnami/discourse/vendor/bundle/ruby/2.7.0/gems/rake-13.0.3/lib/rake/application.rb:103:in `load_rakefile'
/opt/bitnami/discourse/vendor/bundle/ruby/2.7.0/gems/rake-13.0.3/lib/rake/application.rb:82:in `block in run'
/opt/bitnami/discourse/vendor/bundle/ruby/2.7.0/gems/rake-13.0.3/lib/rake/application.rb:186:in `standard_exception_handling'
/opt/bitnami/discourse/vendor/bundle/ruby/2.7.0/gems/rake-13.0.3/lib/rake/application.rb:80:in `run'
/opt/bitnami/discourse/vendor/bundle/ruby/2.7.0/gems/rake-13.0.3/exe/rake:27:in `<top (required)>'
/opt/bitnami/discourse/vendor/bundle/ruby/2.7.0/bin/rake:23:in `load'
/opt/bitnami/discourse/vendor/bundle/ruby/2.7.0/bin/rake:23:in `<top (required)>'
/opt/bitnami/ruby/lib/ruby/gems/2.7.0/gems/bundler-2.2.22/lib/bundler/cli/exec.rb:63:in `load'
/opt/bitnami/ruby/lib/ruby/gems/2.7.0/gems/bundler-2.2.22/lib/bundler/cli/exec.rb:63:in `kernel_load'
/opt/bitnami/ruby/lib/ruby/gems/2.7.0/gems/bundler-2.2.22/lib/bundler/cli/exec.rb:28:in `run'
/opt/bitnami/ruby/lib/ruby/gems/2.7.0/gems/bundler-2.2.22/lib/bundler/cli.rb:474:in `exec'
/opt/bitnami/ruby/lib/ruby/gems/2.7.0/gems/bundler-2.2.22/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/opt/bitnami/ruby/lib/ruby/gems/2.7.0/gems/bundler-2.2.22/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
/opt/bitnami/ruby/lib/ruby/gems/2.7.0/gems/bundler-2.2.22/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
/opt/bitnami/ruby/lib/ruby/gems/2.7.0/gems/bundler-2.2.22/lib/bundler/cli.rb:30:in `dispatch'
/opt/bitnami/ruby/lib/ruby/gems/2.7.0/gems/bundler-2.2.22/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
/opt/bitnami/ruby/lib/ruby/gems/2.7.0/gems/bundler-2.2.22/lib/bundler/cli.rb:24:in `start'
/opt/bitnami/ruby/lib/ruby/gems/2.7.0/gems/bundler-2.2.22/exe/bundle:49:in `block in <top (required)>'
/opt/bitnami/ruby/lib/ruby/gems/2.7.0/gems/bundler-2.2.22/lib/bundler/friendly_errors.rb:128:in `with_friendly_errors'
/opt/bitnami/ruby/lib/ruby/gems/2.7.0/gems/bundler-2.2.22/exe/bundle:37:in `<top (required)>'
/opt/bitnami/ruby/bin/bundle:23:in `load'
/opt/bitnami/ruby/bin/bundle:23:in `<main>'
root@fef60997239b:/opt/bitnami/discourse#
```

**Describe the results you expected:**

Setup to complete...

**Additional information you deem important (e.g. issue happens only occasionally):**

Here is the docker-compose.yml I have (the only thing we changed between fail and success was the discourse tag). It's a setup that connects to a traefik proxy:

```
# See https://github.com/bitnami/bitnami-docker-discourse
version: '3.7'
services:
  discourse:
    image: 'docker.io/bitnami/discourse:2-debian-10'
    labels:
      - 'traefik.enable=true'
      - 'traefik.port=3000'
      - 'traefik.frontend.rule=Host:$DISCOURSE_HOSTNAME'
      - 'traefik.docker.network=tripleperformance_traefik'
    depends_on:
      - postgresql
      - redis
    networks:
      - default
      - tripleperformance_traefik
    volumes:
      - 'discourse_data:/bitnami'
    environment:
      - POSTGRESQL_HOST=postgresql
      - POSTGRESQL_ROOT_USER=postgres
      - POSTGRESQL_CLIENT_CREATE_DATABASE_NAME=bitnami_application
      - POSTGRESQL_CLIENT_CREATE_DATABASE_USERNAME=bn_discourse
      - POSTGRESQL_CLIENT_CREATE_DATABASE_PASSWORD=bitnami1
      - DISCOURSE_POSTGRESQL_NAME=bitnami_application
      - DISCOURSE_POSTGRESQL_USERNAME=bn_discourse
      - DISCOURSE_POSTGRESQL_PASSWORD=bitnami1
      - POSTGRESQL_CLIENT_POSTGRES_NAME=bitnami_application
      - POSTGRESQL_CLIENT_POSTGRES_USERNAME=bn_discourse
      - POSTGRESQL_CLIENT_POSTGRES_PASSWORD=bitnami1      
      - DISCOURSE_HOSTNAME
      - DISCOURSE_ENV=production
      - RAILS_ENV=production

  sidekiq:
    image: 'docker.io/bitnami/discourse:2-debian-10'
    depends_on:
      - discourse
    volumes:
      - 'discourse_data:/bitnami'
    command: 'nami start --foreground discourse-sidekiq'
    environment:
      - DISCOURSE_POSTGRESQL_NAME=bitnami_application
      - DISCOURSE_POSTGRESQL_USERNAME=bn_discourse
      - DISCOURSE_POSTGRESQL_PASSWORD=bitnami1
      - POSTGRESQL_CLIENT_POSTGRES_NAME=bitnami_application
      - POSTGRESQL_CLIENT_POSTGRES_USERNAME=bn_discourse
      - POSTGRESQL_CLIENT_POSTGRES_PASSWORD=bitnami1      
      - DISCOURSE_HOST=discourse
      - DISCOURSE_PORT=3000
      - ALLOW_EMPTY_PASSWORD=yes
      - DISCOURSE_HOSTNAME
      - DISCOURSE_ENV=production
      - RAILS_ENV=production      

  postgresql:
    image: 'docker.io/bitnami/postgresql:12-debian-10'
    environment:
      - ALLOW_EMPTY_PASSWORD=yes
    volumes:
      - 'postgresql_data:/bitnami/postgresql'

  redis:
    image: 'docker.io/bitnami/redis:6.0-debian-10'
    environment:
      - ALLOW_EMPTY_PASSWORD=yes
    volumes:
      - 'redis_data:/bitnami'

  adminer:
    image: adminer
    environment:
      - ADMINER_DEFAULT_SERVER=postgresql
    networks:
      - default
      - tripleperformance_traefik
    labels:
      - 'traefik.enable=true'
      - 'traefik.port=8080'
      - 'traefik.frontend.rule=Host:$ADMINER_HOSTNAME'
      - 'traefik.docker.network=tripleperformance_traefik'

networks:
  default: ~
  tripleperformance_traefik:
    external: true

volumes:
  postgresql_data:
    driver: local
  redis_data:
    driver: local
  discourse_data:
    driver: local

```

**Version**

- Output of `docker version`:

```
Client: Docker Engine - Community
 Cloud integration: 1.0.17
 Version:           20.10.7
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        f0df350
 Built:             Wed Jun  2 11:56:47 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.7
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       b0f5bc3
  Built:            Wed Jun  2 11:54:58 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.6
  GitCommit:        d71fcd7d8303cbf684402823e425e9dd2e99285d
 runc:
  Version:          1.0.0-rc95
  GitCommit:        b9ee9c6314599f1b4a7f497e1f1f856fe433d3b7
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

- Output of `docker info`:

```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Build with BuildKit (Docker Inc., v0.5.1-docker)
  compose: Docker Compose (Docker Inc., v2.0.0-beta.6)
  scan: Docker Scan (Docker Inc., v0.8.0)
WARNING: Plugin "/usr/libexec/docker/cli-plugins/docker-app" is not valid: failed to fetch metadata: fork/exec /usr/libexec/docker/cli-plugins/docker-app: no such file or directory

Server:
 Containers: 21
  Running: 14
  Paused: 0
  Stopped: 7
 Images: 48
 Server Version: 20.10.7
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: d71fcd7d8303cbf684402823e425e9dd2e99285d
 runc version: b9ee9c6314599f1b4a7f497e1f1f856fe433d3b7
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 5.4.72-microsoft-standard-WSL2
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 5.804GiB
 Name: docker-desktop
 ID: KRFG:WRZA:CLL2:N5CL:WGPZ:7R34:46CW:CHHA:LTV6:AJGT:E2S3:2KWX
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No blkio throttle.read_bps_device support
WARNING: No blkio throttle.write_bps_device support
WARNING: No blkio throttle.read_iops_device support
WARNING: No blkio throttle.write_iops_device support
```

- Output of `docker-compose version` (if applicable):

```
docker-compose version 1.29.2, build 5becea4c
docker-py version: 5.0.0
CPython version: 3.7.10
OpenSSL version: OpenSSL 1.1.0l  10 Sep 2019
```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**

Docker + docker-compose on windows (WSL2)

**Comment by Mauraza** [comment]
Hi @bertrandgorge,

Could you try this change 🔽  is the command of sidekiq?
```yaml
 sidekiq:
    image: 'docker.io/bitnami/discourse:2-debian-10'
    depends_on:
      - discourse
    volumes:
      - 'discourse_data:/bitnami'
    command: 'nami start --foreground discourse-sidekiq'
```
to 
```yaml
** sidekiq:
    image: 'docker.io/bitnami/discourse:2-debian-10'
    depends_on:
      - discourse
    volumes:
      - 'discourse_data:/bitnami'
    command: /opt/bitnami/scripts/discourse-sidekiq/run.sh
```

**Comment by bertrandgorge** [comment]
Hello @Mauraza - thanks for looking into my problem...! The change has no effect. Using the latest image + the sidekick command, the install still fails during the database migrations (rake db:migrate)... I'm available for more information if useful !

**Comment by Mauraza** [comment]
Hi @bertrandgorge,

I think your issue is the same that https://github.com/bitnami/bitnami-docker-discourse/issues/199

**Comment by bertrandgorge** [comment]
Hello @Mauraza - yes I saw that - but I don't have any specific plugin.

I realized my docker-compose.yml  was out of date since the 2.7 refact. This is actually a good lead. Let me do some more test starting from the latest bitnami docker-compose.yml, and build my way up to my setup to find what's failing. I'll close the issue for the moment and reopen it if I find that the culprit is clearly identified !!
